korge/src/commonMain/kotlin/com/soywiz/korge/scene/Scene.kt

```diff
@@ -12,41 +12,106 @@ import com.soywiz.korma.geom.*
 import kotlinx.coroutines.*
 import kotlin.coroutines.*
 
+/**
+ * Acts as a controller. Subclasses must override at least one of: [sceneInit] or [sceneMain].
+ *
+ * The lifecycle of the [Scene] for ([SceneContainer.pushTo], [SceneContainer.changeTo], [SceneContainer.back], [SceneContainer.forward]):
+ *
+ * - ## OLD Scene -> NEW Scene -- Scene is changed
+ * - NEW: [sceneInit] - BLOCK - Called when this scene enters. Waits until completed
+ * - NEW: [sceneMain] - DO NOT BLOCK - Called after [sceneInit]. Doesn't wait for completion. Its underlying Job is closed automatically on [sceneAfterDestroy]
+ * - OLD: [sceneBeforeLeaving] - BLOCK - Called on the old scene. Waits until completed.
+ * - Transition from the old view and the new view is performed [SceneContainer.transitionView] [View.ratio] 0..1
+ * - OLD: [sceneDestroy] - BLOCK - Called when this scene exits. Waits until completed.
+ * - OLD: [sceneAfterDestroy] - DO NOT BLOCK - Cancels the [Job] of the old scene. Do not wait
+ * - NEW: [sceneAfterInit] - DO NOT BLOCK - Similar to [sceneMain] but after the transition.
+ * - ## New scene is returned
+ */
 abstract class Scene : InjectorAsyncDependency, ViewsContainer, CoroutineScope {
+    /** A child [AsyncInjector] for this instance. Set by the [init] method. */
 	lateinit var injector: AsyncInjector
+    /** The [Views] singleton of the application. Set by the [init] method. */
 	override lateinit var views: Views
-	val ag: AG get() = views.ag
+    /** The [SceneContainer] [View] that contains this [Scene]. Set by the [init] method. */
 	lateinit var sceneContainer: SceneContainer
+    /** The [ResourcesRoot] singleton. Set by the [init] method. */
 	lateinit var resourcesRoot: ResourcesRoot
-	//protected lateinit var bus: Bus
-	internal val _sceneViewContainer: Container = Container()
+
+    /** A reference to the [AG] from the [Views] */
+    val ag: AG get() = views.ag
+
+    /** A [Container] view that */
+    internal val _sceneViewContainer: Container = Container()
+
 	val root get() = _sceneViewContainer
+
 	protected val cancellables = CancellableGroup()
-	override val coroutineContext: CoroutineContext by lazy { Job(views.coroutineContext[Job.Key]) }
+	override val coroutineContext: CoroutineContext = Job(views.coroutineContext[Job.Key])
 	val sceneView: Container = createSceneView().apply {
 		_sceneViewContainer += this
 	}
 	protected open fun createSceneView(): Container = Container()
 
-	override suspend fun init(injector: AsyncInjector): Unit {
+    /**
+     * This method will be called by the [SceneContainer] that will display this [Scene].
+     * This will set the [injector], [views], [sceneContainer] and [resourcesRoot].
+     */
+    final override suspend fun init(injector: AsyncInjector) {
```

Comment:
init is flagged as deprecated

Reply:
Where?

Reply:
Here

https://github.com/korlibs/korge/blob/b3a8af5c2a2e5db8ae808629a89e970d875b6750/korge/src/commonMain/kotlin/com/soywiz/korge/scene/Module.kt#L40

Reply:
Oh I see. Not that `init`.
This `init` https://github.com/korlibs/korinject/blob/c81a6e9c04a16c4e9ad14573ea792b2d1fdcde63/korinject/src/commonMain/kotlin/com/soywiz/korinject/AsyncInjector.kt#L202